hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -1167,7 +1216,10 @@ public String getBucketLocation(String bucketName) throws IOException {
     final String region = trackDurationAndSpan(
         STORE_EXISTS_PROBE, bucketName, null, () ->
             invoker.retry("getBucketLocation()", bucketName, true, () ->
-                s3.getBucketLocation(bucketName)));
+                // If accessPoint then region is known from Arn
+                accessPoint != null
```

Comment:
Should we pull this up to L1216 & we can skip the entire overhead of duration tracking, retry etc.

Currently it's overkill to wrap, but it will add it to the iostats, so maybe it's best to leave as is

Reply:
I liked the iostats tracking so it doesn't look like an operation is missing / changed.